### PR TITLE
test(crap4rs): #434 — property tests for the LCOV path-normalization seam

### LINKED ISSUE #434: test: add property test for LCOV normalize_path I/O seam (Boundary-Rule deferral)
## Summary
Boundary-Rule accountability for the LCOV parser's exclusion from the fuzz epic: its `.*` panic-freedom is already proptested, so it is correctly NOT fuzzed. The one residual signal — the `normalize_path` / `.is_file()` I/O seam with adversarial `..`/long-suffix paths interacting with root_path — belongs at the PROPERTY-TEST level, not fuzz. Add that property test.

## Acceptance Criteria
- [ ] A proptest over adversarial path inputs to the LCOV `normalize_path`/suffix-match seam (no panic; traversal guard holds; no cross-root leakage).
- [ ] Lives alongside the existing crap4rs coverage proptests; commits any generated regression file.

## Context
- Epic: #428 (captures the LCOV deferral so the exclusion is tracked, per the exclusions-accountability rule).
- Seam: `normalize_path` in crap4rs coverage/mod.rs (the one I/O-bearing path; already has a single example test).

## Discovery
- Whether to reuse a tempdir harness for the `.is_file()` probe or assert purely on the lexical suffix-match.

### PR DESCRIPTION
**Pull request by cmbays**
## Summary
Boundary-Rule deferral accountability for excluding the LCOV parser from the Q4 fuzz epic (#428). The LCOV parser's `.*` panic-freedom is **already** proptested, so fuzzing it would duplicate a proven invariant. Its one residual signal — the I/O-bearing `normalize_path` / `suffix_match_under` path handling — belongs at the **property** level, not fuzz. **Independent of the fuzz stack** (branches off `main`, no bolero).

## What's here
Two proptests in the existing inline `mod proptests` (crap4rs `adapters/coverage/mod.rs`):
- `normalize_path_never_panics` — panic-freedom over arbitrary SF-path input under an arbitrary root (adversarial `..`, mixed separators, unicode, long suffixes).
- `suffix_match_under_is_rooted_and_parentdir_free` — the structural traversal guard: when a real file resolves, the returned suffix has no `..` component and always re-anchors under the root. A real tempdir file + `..`/junk-prefixed input keeps the `Some` branch non-vacuous.

Complements the existing example-based traversal unit tests (`parentdir_sf_path_does_not_take_fast_path_is_file_probe`, etc.).

## Gates (all green locally)
fmt · clippy-1.96 -D warnings · nextest --workspace --all-targets (incl. both new proptests) · MSRV 1.93 · doc -D warnings · mutants-skip lint · **strict-8 self-gate: 1681 fns, 0 above threshold, PASS**. No new deps (proptest + tempfile already dev-deps). The `proptest!` block is at the file's end and its macro body isn't walked as individual functions → self-CRAP set + `wire_envelope_crap4rs` snapshot unchanged.

Closes #434

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive property-based unit tests to expand testing coverage of path-handling functionality. These new tests validate system stability across diverse input scenarios, including edge cases and unexpected inputs, strengthening code robustness and helping prevent potential failures in various operating conditions.

### CODE REVIEW

File: crates/crap4rs/src/adapters/coverage/mod.rs [hunk 1]
```diff
@@ -1011,6 +1011,66 @@ mod proptests {
...
+        ) {
+            use std::io::Write;
+            let dir = tempfile::tempdir().unwrap();
+            let root = dir.path();
+            let leaf_dir = root.join("pkg");
+            std::fs::create_dir_all(&leaf_dir).unwrap();
+            std::fs::File::create(leaf_dir.join("leaf.rs"))
+                .unwrap()
+                .write_all(b"fn x() {}")
+                .unwrap();
```
[the remark this answers is not in the document]
Comment (cmbays) [reply crates/crap4rs/src/adapters/coverage/mod.rs]: Applied in 74845d2 — moved the tempdir + file creation into a `OnceLock<TempDir>` initialized once for the whole run (the test only reads it), removing the per-case I/O. Thanks.

File: crates/crap4rs/src/adapters/coverage/mod.rs [hunk 2]
```diff
@@ -1011,6 +1011,66 @@ mod proptests {
...
+                    !Path::new(&suffix)
+                        .components()
+                        .any(|c| c == std::path::Component::ParentDir),
+                    "suffix_match_under returned a `..`-bearing suffix: {suffix}"
+                );
+                prop_assert!(
+                    root.join(&suffix).is_file(),
+                    "suffix {suffix} does not re-anchor to a real file under root"
+                );
+            }
```
[the remark this answers is not in the document]
Comment (cmbays) [reply crates/crap4rs/src/adapters/coverage/mod.rs]: Good catch — fixed in 74845d2. The crafted path always has a real `pkg/leaf.rs` leaf so a match is guaranteed; now `prop_assert!(suffix.is_some(), …)` first, then validate the guard on the unwrapped suffix. A regression of `suffix_match_under` to always-`None` now fails the test instead of passing vacuously.
